node/nodeTransactions_test.go

```diff
@@ -294,6 +296,97 @@ func TestNode_GetTransaction_ShouldNotFindAndReturnUnknown(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestNode_ComputeTransactionStatusAllBranches(t *testing.T) {
+	t.Parallel()
+
+	throttler := &mock.ThrottlerStub{
+		CanProcessCalled: func() bool {
+			return true
+		},
+	}
+	dataPool := &mock.PoolsHolderStub{
+		TransactionsCalled:         getCacherHandler(false, ""),
+		RewardTransactionsCalled:   getCacherHandler(false, ""),
+		UnsignedTransactionsCalled: getCacherHandler(false, ""),
+	}
+	storer := &mock.ChainStorerMock{
+		GetStorerCalled: func(unitType dataRetriever.UnitType) storage.Storer {
+			return getStorerStub(false)
+		},
+	}
+
+	shardZeroAddr := []byte("addrShard0")
+	shardOneAddr := []byte("addrShard1")
+	shardCoordinator := &mock.ShardCoordinatorMock{
+		ComputeIdCalled: func(addr []byte) uint32 {
+			if bytes.Equal(shardZeroAddr, addr) {
+				return 0
+			}
+			return 1
+		},
+	}
+
+	n, _ := node.NewNode(
+		node.WithApiTransactionByHashThrottler(throttler),
+		node.WithDataPool(dataPool),
+		node.WithDataStore(storer),
+		node.WithShardCoordinator(shardCoordinator),
+	)
+
+	rwdTxCrossShard := &rewardTx.RewardTx{RcvAddr: shardZeroAddr}
+	normalTxIntraShard := &transaction.Transaction{RcvAddr: shardZeroAddr, SndAddr: shardZeroAddr}
+	normalTxCrossShard := &transaction.Transaction{RcvAddr: shardOneAddr, SndAddr: shardZeroAddr}
+	unsignedTxIntraShard := &smartContractResult.SmartContractResult{RcvAddr: shardOneAddr, SndAddr: shardZeroAddr}
```

Comment:
This one is cross shard. Also add another one for intra-shard :)

Reply:
done.